src/main/java/net/schmizz/sshj/transport/TransportImpl.java

```diff
@@ -658,17 +661,31 @@ ConnInfo getConnInfo() {
     }
 
     @Override
-    public KeyAlgorithm getKeyAlgorithm(KeyType keyType) throws TransportException {
-        for (KeyAlgorithm ka : keyAlgorithms) {
-            if (ka.getKeyFormat().equals(keyType)) {
-                return ka;
-            }
-        }
+    public void setHostKeyAlgorithm(KeyAlgorithm keyAlgorithm) {
+        this.hostKeyAlgorithm = keyAlgorithm;
+    }
 
-        throw new TransportException("Cannot find an available KeyAlgorithm for type " + keyType);
+    @Override
+    public KeyAlgorithm getHostKeyAlgorithm() {
+        return this.hostKeyAlgorithm;
     }
 
-    public void setKeyAlgorithms(List<KeyAlgorithm> keyAlgorithms) {
-        this.keyAlgorithms = keyAlgorithms;
+    @Override
+    public void setRSASHA2Support(Boolean rsaSHA2Support) {
+        this.rsaSHA2Support = rsaSHA2Support;
+    }
+
+    @Override
+    public KeyAlgorithm getClientKeyAlgorithm(KeyType keyType) throws TransportException {
```

Comment:
Can we add a test that failed with the old implementation and now succeeds. That will prevent regression

Reply:
I have a test, but it unearthed an unrelated issue with `ecdsa-sha2-nistp521` keys. I will see whether I can fix it.

Reply:
I have added tests that fail with the old implementation and succeed with the new one. However, this requires https://github.com/hierynomus/sshj/pull/623, which fixes the incorrect encoding of long ECDSA signatures. Please merge that one first for the tests to pass.